src/app/event/[id]/EventDetail.tsx

```diff
@@ -1,18 +1,103 @@
-import { getEvent } from '../../../api/events';
+'use client';
+
+import format from 'date-fns/format';
+import cx from 'classnames';
+import { useState } from 'react';
+
+import { HeartIcon } from '../../../components/icons/HeartIcon';
+import { mockEvent } from './mock';
+import { Event } from '../../../types/Event';
 
 type EventDetailProps = {
-  id: string;
+  event: Event;
+  className?: string;
 };
 
-export const EventDetail: React.FC<EventDetailProps> = async ({ id }) => {
-  const event = await getEvent(id);
+export const EventDetail: React.FC<EventDetailProps> = ({
+  event,
+  className = '',
+}) => {
+  const [isLiked, setIsLiked] = useState(false);
+
+  const handleLikeEvent = () => {
+    setIsLiked(!isLiked);
+  };
```

Comment:
Move this to heart component and you can still keep this as server component

Reply:
This can not be server component soon anyway 😆, but bood idea... i will use heart  more often maybe.